scipy/io/_mmio.py

```diff
@@ -20,6 +22,24 @@
 
 __all__ = ['mminfo', 'mmread', 'mmwrite', 'MMFile']
 
+"""
+Maximum number of digits for `mmwrite` function.
+This can be calculated by below python codes:
+eps = np.finfo(np.float64).eps
+max_precision = np.floor(-np.log10(eps)).astype(int)
+"""
+MAX_PRECISION = 15
+
+
+def _validate_precision(precision):
+    if precision is not None:
+        # check the input can be converted to an integer
+        precision = operator.index(precision)
+        if not (1 <= precision <= MAX_PRECISION):
+            msg = (f"From SciPy 1.18.0, an exception will be thrown if the "
+                   f"precision input is outside the range of 1 "
+                   f"to {MAX_PRECISION-1}")
```

Comment:
```suggestion
                   f"to {MAX_PRECISION}")
```
The check is not a strict inequality?

Reply:
Good catch!!